aten/src/ATen/native/cuda/DistanceKernel.cu

```diff
@@ -270,10 +301,50 @@ void pdist_backward_kernel_impl(Tensor& result, const Tensor& grad, const Tensor
   at::sum_out(result, buffer, 0);
 }
 
+void cdist_backward_kernel_impl(Tensor& result, const Tensor& grad, const Tensor& x1, const Tensor& x2, const double p, const Tensor& dist) {
+  if (p == 0.0 || grad.numel() == 0 || x1.numel() == 0 || x2.numel() == 0) {
+    result.fill_(0);
+    return;
+  }
+
+  const int64_t r1 = x1.size(-2);
+  const int64_t r2 = x2.size(-2);
+  const int64_t m = x1.size(-1);
+  const int block_x = 64;
+  const int block_y = 16;
+  const int grid_x = (m + block_x * 8 - 1) / (block_x * 8);
+  const int grid_y = (dist.numel() + block_y - 1) / block_y;
+
+  const dim3 grid(grid_x, grid_y);
+  const dim3 block(block_x, block_y);
+
+  const int64_t count = dist.numel();
+
+  Tensor buffer = at::empty({r2, r1, m}, result.options());
+  AT_DISPATCH_FLOATING_TYPES(result.type(), "cdist_cuda_backward", [&] {
+    if (p == 1.0) {
+      cdist_backward_kernel_cuda_impl<scalar_t, dists<scalar_t>::one><<<grid, block>>>(buffer.data<scalar_t>(), grad.data<scalar_t>(), x1.data<scalar_t>(), x2.data<scalar_t>(), dist.data<scalar_t>(), grad.stride(1), p, r1, r2, m, count);
+    } else if (p < 2.0) {
+      cdist_backward_kernel_cuda_impl<scalar_t, dists<scalar_t>::lt_two><<<grid, block>>>(buffer.data<scalar_t>(), grad.data<scalar_t>(), x1.data<scalar_t>(), x2.data<scalar_t>(), dist.data<scalar_t>(), grad.stride(1), p, r1, r2, m, count);
+    } else if (p == 2.0) {
+      cdist_backward_kernel_cuda_impl<scalar_t, dists<scalar_t>::two><<<grid, block>>>(buffer.data<scalar_t>(), grad.data<scalar_t>(), x1.data<scalar_t>(), x2.data<scalar_t>(), dist.data<scalar_t>(), grad.stride(1), p, r1, r2, m, count);
+    } else if (std::isinf(p)) {
+      cdist_backward_kernel_cuda_impl<scalar_t, dists<scalar_t>::inf><<<grid, block>>>(buffer.data<scalar_t>(), grad.data<scalar_t>(), x1.data<scalar_t>(), x2.data<scalar_t>(), dist.data<scalar_t>(), grad.stride(1), p, r1, r2, m, count);
+    } else {
+      cdist_backward_kernel_cuda_impl<scalar_t, dists<scalar_t>::p><<<grid, block>>>(buffer.data<scalar_t>(), grad.data<scalar_t>(), x1.data<scalar_t>(), x2.data<scalar_t>(), dist.data<scalar_t>(), grad.stride(1), p, r1, r2, m, count);
+    }
+  });
+  AT_CUDA_CHECK(cudaGetLastError());
+
+  at::sum_out(result, buffer, 0);
+}
+
+
 } // anonymous namespace
 
 REGISTER_DISPATCH(pdist_forward_stub, &pdist_forward_kernel_impl);
 REGISTER_DISPATCH(pdist_backward_stub, &pdist_backward_kernel_impl);
 REGISTER_DISPATCH(cdist_stub, &cdist_kernel_impl);
+REGISTER_DISPATCH(cdist_backward_stub, &cdist_backward_kernel_impl);
 
-}} // at::native
+}} // at::native
```

Comment:
nit: I think there should be a newline here.

Reply:
Fixed